### PR TITLE
Remove deprecated DESTDIR and unused things from install.rb

### PR DESCRIPTION
**Pull request by ekohl**
I was trying to figure out how to do the install parts for https://github.com/puppetlabs/puppet/pull/8636 and trying to make sense of the install script. In doing so I found some things that have been unused for ages. Things get easier to understand if there's less stuff, so this helps a bit.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

File: install.rb [hunk 1]
```diff
@@ -204,16 +180,6 @@ def prepare_installation
     opts.on('--no-batch-files', 'Prevents installation of batch files for windows', 'Default off') do |batch_files|
       InstallOptions.batch_files = false
     end
-    opts.on('--quick', 'Performs a quick installation. Only the', 'installation is done.') do |quick|
-      InstallOptions.rdoc    = false
-      InstallOptions.ri      = false
-      InstallOptions.configs = true
-    end
```
Comment (joshcooper) [inline install.rb:211]: We use `--quick` when building puppet-agent packages. Here's where that happens https://github.com/puppetlabs/puppet-agent/blob/b7976e5c12a8816f00d48a45266ab0f27ebef379/configs/components/puppet.rb#L145-L161

Comment (ekohl) [reply install.rb:211]: Ok, but it does nothing now. Do you prefer to keep it in as a dummy option, raising a warning?

Comment (joshcooper) [reply install.rb:211]: yes a noop is fine with a warning to stderr

Comment (ekohl) [reply install.rb:211]: I kept in `configs = true` since technically that does something, even though it's on by default already.

**Comment by joshcooper** [comment]
jenkins please test this

**Review by joshcooper — APPROVED** [review]

**Comment by joshcooper** [comment]
just waiting on tests

**Pull request merged**